client/src/utils/authFetch.js

```diff
@@ -0,0 +1,15 @@
+export async function getProtectedData(path) {
+  try {
+    const response = await fetch(path, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!response.ok) throw new Error(response.status);
+    const json = await response.json();
+    return json;
```

Comment:
You still need to check if the response is 401 and handle that deleting the user obj and redirecting